internal/security/proxy/requestor.go

```diff
@@ -17,15 +17,37 @@ package proxy
 
 import (
 	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
 	"net/http"
 	"time"
 
 	"github.com/edgexfoundry/edgex-go/internal"
 )
 
 func NewRequestor(skipVerify bool, timeoutInSecond int) internal.HttpCaller {
-	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
+	var tr *http.Transport
+	if skipVerify {
+		tr = &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
+		}
+	} else {
+		caCert, err := ioutil.ReadFile(Configuration.SecretService.CACertPath)
+		if err != nil {
+			LoggingClient.Error("failed to load rootCA certificate.")
+			return nil
+		}
+		LoggingClient.Info("successfully loaded rootCA certificate.")
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+
+		tr = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:            caCertPool,
+				InsecureSkipVerify: skipVerify,
```

Comment:
minor nitpick you could just drop this line since in this block we already know that it's false and the default value for a bool in Go is false so we don't gain anything by having this line and IMHO it just opens us up to accidental typos or something modifying this here when we shouldn't be